gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/SpecimenCompoundsMapper.java

```diff
@@ -0,0 +1,224 @@
+            .filter(RCMRMT030101UK04Component02::hasObservationStatement)
+            .map(RCMRMT030101UK04Component02::getObservationStatement)
+            .forEach(observationStatement -> {
+                var observationOpt = getObservationById(observations, observationStatement.getId().getRoot());
+                observationOpt.ifPresent(observation -> handleObservationStatement(
+                    specimenCompoundStatement, observation, diagnosticReport)
+                );
+                handleNarrativeStatements(clusterCompoundStatement, observationComments, observationOpt.orElse(null));
+            });
+    }
+
+    private void handleBatteryCompoundStatement(RCMRMT030101UK04CompoundStatement specimenCompoundStatement,
+        RCMRMT030101UK04CompoundStatement batteryCompoundStatement,
+        List<Observation> observations, List<Observation> observationComments, DiagnosticReport diagnosticReport) {
+
+        batteryCompoundStatement.getComponent().stream()
+            .filter(RCMRMT030101UK04Component02::hasCompoundStatement)
+            .map(RCMRMT030101UK04Component02::getCompoundStatement)
+            .filter(compoundStatement -> CLUSTER_CLASSCODE.equals(compoundStatement.getClassCode().get(0)))
+            .forEach(compoundStatement -> handleClusterCompoundStatement(
+                specimenCompoundStatement, compoundStatement, observations, observationComments, diagnosticReport
+            ));
+
+        batteryCompoundStatement.getComponent().stream()
+            .filter(RCMRMT030101UK04Component02::hasCompoundStatement)
+            .map(RCMRMT030101UK04Component02::getCompoundStatement)
+            .forEach(compoundStatement -> compoundStatement.getComponent().stream()
+                .filter(RCMRMT030101UK04Component02::hasObservationStatement)
+                .map(RCMRMT030101UK04Component02::getObservationStatement)
+                .findFirst()
+                .flatMap(observationStatement -> getObservationById(observations, observationStatement.getId().getRoot()))
+                .ifPresent(observation -> handleObservationStatement(
+                    specimenCompoundStatement, observation, diagnosticReport)
+                ));
+
+        batteryCompoundStatement.getComponent().stream()
+            .filter(RCMRMT030101UK04Component02::hasObservationStatement)
+            .map(RCMRMT030101UK04Component02::getObservationStatement)
+            .forEach(observationStatement ->
+                getObservationById(observations, observationStatement.getId().getRoot()).ifPresent(observation -> {
+                    handleObservationStatement(specimenCompoundStatement, observation, diagnosticReport);
+                    handleNarrativeStatements(batteryCompoundStatement, observationComments, observation);
+                }));
+    }
+
+    private Optional<RCMRMT030101UK04CompoundStatement> getCompoundStatementByDRId(RCMRMT030101UK04EhrExtract ehrExtract, String id) {
```

Comment:
Similar to comment on specimen mapper, nested components can cause NPE's try using extractAllCompoundStatement methods when searching at this level